Mathlib/Topology/Algebra/Ring/Basic.lean

```diff
@@ -81,6 +81,43 @@ instance (priority := 50) DiscreteTopology.topologicalRing [TopologicalSpace α]
 
 section
 
+namespace NonUnitalSubsemiring
+
+variable [TopologicalSpace α] [NonUnitalSemiring α] [TopologicalSemiring α]
+
+instance instTopologicalSemiring (S : NonUnitalSubsemiring α) : TopologicalSemiring S :=
+  { S.toSubsemigroup.continuousMul, S.toAddSubmonoid.continuousAdd with }
+
+/-- The (topological) closure of a non-unital subsemiring of a non-unital topological semiring is
+itself a non-unital subsemiring. -/
+def topologicalClosure (s : NonUnitalSubsemiring α) : NonUnitalSubsemiring α :=
+  { s.toSubsemigroup.topologicalClosure, s.toAddSubmonoid.topologicalClosure with
+    carrier := _root_.closure (s : Set α) }
+
+@[simp]
+theorem topologicalClosure_coe (s : NonUnitalSubsemiring α) :
+    (s.topologicalClosure : Set α) = _root_.closure (s : Set α) :=
+  rfl
+
+theorem le_topologicalClosure (s : NonUnitalSubsemiring α) : s ≤ s.topologicalClosure :=
+  _root_.subset_closure
+
+theorem isClosed_topologicalClosure (s : NonUnitalSubsemiring α) :
+    IsClosed (s.topologicalClosure : Set α) := isClosed_closure
+
+theorem topologicalClosure_minimal (s : NonUnitalSubsemiring α) {t : NonUnitalSubsemiring α}
+    (h : s ≤ t) (ht : IsClosed (t : Set α)) : s.topologicalClosure ≤ t :=
+  closure_minimal h ht
+
+/-- If a non-unital subsemiring of a non-unital topological semiring is commutative, then so is its
+topological closure. -/
+def nonUnitalCommSemiringTopologicalClosure [T2Space α] (s : NonUnitalSubsemiring α)
```

Comment:
Should these be `abbrev`s as well? It seems that there are a lot of these in the library so don't bother changing them all in this PR.

Reply:
yes, thanks, good catch.